aten/src/ATen/gen.py

```diff
@@ -451,19 +442,19 @@ def declare_outputs():
             if kind != 'Type' and density == "Sparse":
                 # No Storage or Tensor for sparse
                 continue
-            fm.will_write("{}{}{}.h".format(full_backend, scalar_name, kind))
-            fm.will_write("{}{}{}.cpp".format(full_backend, scalar_name, kind))
-        # output LegacyTHDispatchers
+            fm.will_write("{}{}.h".format(full_backend, kind))
+            fm.will_write("{}{}.cpp".format(full_backend, kind))
+    # output LegacyTHDispatchers
+    for backend, density, scalar_type in legacy_iterate_types():
+        fm = file_manager
+        if backend == 'CUDA':
+            fm = cuda_file_manager
         if density == 'Dense':
-            fm.will_write("{}{}{}{}.h".format('LegacyTH', full_backend, scalar_name, 'Dispatcher'))
-            fm.will_write("{}{}{}{}.cpp".format('LegacyTH', full_backend, scalar_name, 'Dispatcher'))
+            fm.will_write("{}{}{}{}.h".format('LegacyTH', backend, scalar_type[0], 'Dispatcher'))
```

Comment:
what is this LegacyTHDispatcher? it seems to be a bunch of empty cpp classes. Can't we just nuke all of that now?

Reply:
@li-roy did you answer this?

This was meant to move all the TH/THNN functions to a different mechanism, primarily because they aren't end-user methods (and don't have a schema).  The unsolved problem is how to handle autogrrad: they still need variable type definitions that know calls to them need to be unwrapped.